src/fides/api/ops/api/v1/endpoints/policy_endpoints.py

```diff
@@ -443,6 +449,29 @@ def get_rule_target(
     return get_rule_target_or_error(db, policy_key, rule_key, rule_target_key)
 
 
+def _validate_data_categories(
+    db: Session,
+    data_categories: List[str],
+) -> None:
+    """
+    Helper method to validate that all data categories supplied to the endpoint
+    exist in the database
+    """
+    from_db = [
+        dc.fides_key
+        for dc in DataCategory.filter(
+            db=db,
+            conditions=DataCategory.fides_key.in_(data_categories),
+        ).all()
+    ]
+    invalid_categories = list(set(data_categories) - set(from_db))
+    if invalid_categories:
+        raise HTTPException(
+            status_code=HTTP_422_UNPROCESSABLE_ENTITY,
+            detail=f"Invalid data categories: {invalid_categories}",
+        )
```

Comment:
I think model-level validation is sufficient here?

Reply:
I added this to best-replicate the previous experience where data is validated before we attempt to store it in the DB, rather than by the DB.

Reply:
Well it isn't being validated by the db specifically, it's being validated before it gets there by the create/update methods.  And the rule target endpoint has a try except specifically to catch the DataCategoryNotSupported error that is now superfluous.

So we are running identical checks back-to-back here, but you can leave this if you prefer.